src/components/workspace/MCQChooser.tsx

```diff
@@ -53,16 +47,49 @@ class MCQChooser extends React.PureComponent<IMCQChooserProps, State> {
    * and mcq submission with a given answer id.
    *
    * Post-condition: the local state will be updated to store the
-   * mcq option selected.
+   * mcq option selected, and a notification will be displayed with
+   * a hint, if the question is ungraded.
    *
    * @param i the id of the answer
    */
   private onButtonClickFactory = (i: number) => (e: any) => {
     this.props.handleMCQSubmit(i)
+    if (this.props.mcq.solution && i === this.props.mcq.solution) {
+      showSuccessMessage(this.props.mcq.choices[i].hint!, 4000)
+    } else if (this.props.mcq.solution && i !== this.props.mcq.solution) {
+      showWarningMessage(this.props.mcq.choices[i].hint!, 4000)
+    }
     this.setState({
       mcqOption: i
     })
   }
+
+  /**
+   * Handles the logic for what intent an MCQ option should show up as.
+   * This is dependent on the presence of an actual solution (for ungraded assessments),
+   * the current selection, and whether the selected option is active.
+   *
+   * @param currentOption the current button key, corresponding to a choice ID
+   * @param chosenOption the mcq option that is chosen in the state, i.e what should show up as "selected"
+   * @param solution the solution to the mcq, if any
+   */
+  private getButtonIntent = (
```

Comment:
If possible could you review this with a bit more care? It looks ok to me, but feels like there might be come edge cases I haven't accounted for @ningyuansg